### ISSUE TITLE
Can tracks and wheels of the Bunker URDF rotate in rviz or gazebo?

### ISSUE THREAD

**Issue by djf007** [agilexrobotics/ugv_gazebo_sim#20]
Can tracks and wheels of the Bunker rotate in rviz or gazebo? Thanks!

**Comment by bascetta74** [comment]
Did you find a way to move tracks? Wheels rotate if bunker is tip over (but not tracks), if tracks touch the ground nothing moves.

**Comment by djf007** [comment]
> Did you find a way to move tracks? Wheels rotate if bunker is tip over (but not tracks), if tracks touch the ground nothing moves. Thanks!
No, I can't find a way to solve this problem. This should be caused by its model files.

**Comment by bascetta74** [comment]
I agree, there is a mistake in the original model.
We are working on that... if you are using ROS1 there is a simple solution, in gazebo 8 and 11 you can find an example of a simple tracked vehicle that can be used to correct the original Bunker model (https://answers.ros.org/question/249874/tracked-robot-gazebo-ros/), and you can directly use this plugin https://github.com/lprobsth/gazebo_ros_tracked_vehicle_interface that supports the kinematics of tracked vehicles.

Unfortunately, no porting of that plugin for ROS2 exists... but we are working on it.
Regards,
Luca Bascetta

Luca Bascetta, PhD.
Associate Professor
Chair of the Agricultural Engineering Study Programme

Politecnico di Milano
***@***.***
bascetta.deib.polimi.it<http://bascetta.deib.polimi.it>

[cid:ef5ef52b-5bc4-4adf-8efd-23a79f3fa1fa]


________________________________
Da: djf007 ***@***.***>
Inviato: martedì 28 novembre 2023 09:25
A: agilexrobotics/ugv_gazebo_sim ***@***.***>
Cc: Luca Bascetta ***@***.***>; Comment ***@***.***>
Oggetto: Re: [agilexrobotics/ugv_gazebo_sim] Can tracks and wheels of the Bunker URDF rotate in rviz or gazebo? (Issue #20)


Did you find a way to move tracks? Wheels rotate if bunker is tip over (but not tracks), if tracks touch the ground nothing moves. Thanks!
No, I can't find a way to solve this problem. This should be caused by its model files.
